### ISSUE TITLE
Audio+Video Lag

### ISSUE THREAD

**Issue by vvedharaju** [AgoraIO-Community/AgoraWebSDK-NG#27]
We switched our production app from the old sdk to the ng sdk. We never noticed a mismatch in audio and video tracks with the old sdk, but now we see that pretty often. Is there something we can do to better handle that?

**Comment by disoul** [comment]
There is a known issue about av-sync in v0.1.9. If you call `unpublish` and `publish` for a single track frequently, the audio delay or video delay will increase.

This issue has been fixed in the next version and will be released next week.

**Comment by vvedharaju** [comment]
great, thanks for the quick responses on issues!

**Comment by vvedharaju** [comment]
Actually we are also seeing this even when publish and unpublish are not called frequently. The call starts out fine, after some time  about 2-3 minutes the audio starts playing ahead of the video.

**Comment by vvedharaju** [comment]
The audio plays first and the video plays almost 6 seconds later, even when all 3 people have good internet.

**Comment by vvedharaju** [comment]
Hi, is there an update on when this will be released? We're launching our product this week, and are really looking forward to this fix. Thank you!

**Comment by disoul** [comment]
We are ready to release the next version this Wensday

**Comment by Bioto** [comment]
> We are ready to release the next version this Wensday

@disoul Hey, i'm not sure if this is related or not. But we are running into an issue where some users will not receive audio for 10-60 seconds 50%-70% of the time. We also recieve the following errors in console:

```
AUDIO_INPUT_LEVEL_TOO_LOW_RECOVER
SEND_AUDIO_BITRATE_TOO_LOW_RECOVER
```

We have adjusted the bit rates but are still running into the same issue.

Any ideas?

**Comment by disoul** [comment]
@vvedharaju `v4.0.0` has been released, you can confirm whether this issue is fixed now.

**Comment by vvedharaju** [comment]
@disoul are there any backwards incompatible changes we should be aware of

**Comment by disoul** [comment]
@vvedharaju you can read the release note https://agoraio-community.github.io/AgoraWebSDK-NG/docs/en/release_note

**Comment by vvedharaju** [comment]
@disoul with the new mute/unmute behavior how would we address the scenario where we want to ensure that when the speaker just mutes/unmutes temporarily we still show that they are a part of the conversation. With the new enabled behavior both tracks would be unpublished right?

**Comment by disoul** [comment]
> @disoul with the new mute/unmute behavior how would we address the scenario where we want to ensure that when the speaker just mutes/unmutes temporarily we still show that they are a part of the conversation. With the new enabled behavior both tracks would be unpublished right?

yes, there is no `mute-state` in `v4.0.0`. If you call `setEnabled(false)`, the remote user will receive a `user-unpublished` event.

**Comment by vvedharaju** [comment]
for example, in google meet once I join, even if I mute both my audio and video for one minute because someone is walking by, people still see a box with my name on it. In our live broadcast use case, if one of the hosts just mutes their streams temporarily, we would want to still show the audience that the speaker is there. how do you suggest we handle that use case?

**Comment by disoul** [comment]
You can use `user-joined`/`user-left` to check whether the remote user is still in the live room, use `user-published` / `user-unpublished` to check whether the remote user is enabled his microphone or camera.
.

**Comment by vvedharaju** [comment]
okay will try that out 👍 thanks! Really appreciate how fast responses have been!

**Comment by vvedharaju** [comment]
@disoul the audio video lag hasn't been happening as frequently. it happened to us today on an event, but i'm not sure if it was an sdk issue or an internet issue, but the audio was ahead of the video by a couple of seconds

**Comment by vvedharaju** [comment]
@disoul still noticing this issue

**Comment by vvedharaju** [comment]
@disoul is there any update on this. we are still seeing this issue. there is a frequent mismatch between the audio and video tracks.

**Comment by vvedharaju** [comment]
@disoul previously the issue was from the stream publishing end. the user who was facing lag had to leave and rejoin and that would fix the issue. Now I am facing the issue from the stream playing end. After a while there is a lag between audio and video, and if I perform any user action which causes the stream to stop playing and start playing again, then it readjusts.

**Comment by vvedharaju** [comment]
@HAHAHA44 any update on this issue?

**Comment by vvedharaju** [comment]
@HAHAHA44 @disoul would really appreciate an update on this issue from any one. We still see lag every once in a while and I'm not sure if its still an sdk issue or something else

**Comment by HAHAHA44** [comment]
Thanks for your attention, our gateway has been optimized for audio and video sync，It is testing in grayscale.

**Comment by vvedharaju** [comment]
@HAHAHA44 is the testing done now?

**Comment by vvedharaju** [comment]
@HAHAHA44 any update on this testing?

**Comment by HAHAHA44** [comment]
Av-Sync is online, did you have the issue yet?
